src/main/java/com/example/restapp/dao/AbstractDAO.java

```diff
@@ -4,13 +4,13 @@
 
 public abstract class AbstractDAO {
 
-	protected BeanSession session;
-
-	public void setBeanSession(BeanSession session) {
+	private final BeanSession session;
```

Comment:
There's a reason to use private instead of protected? With protected the subclass can call session directly.. To simplify that, maybe we can change `getBeanSession()` just to `session()`